debezium-connect-rest-extension/src/main/java/io/debezium/kcrestextension/DebeziumResource.java

```diff
@@ -107,40 +83,47 @@ else if (m.lookingAt()) {
+            Herder herder = getHerder();
+            addConnectorPlugins(connectorPlugins, herder.plugins().sinkConnectors());
+            addConnectorPlugins(connectorPlugins, herder.plugins().sourceConnectors());
+            this.availableConnectorPlugins = Collections.unmodifiableList(connectorPlugins);
+        }
     }
 
-    private synchronized List<TransformsInfo> getTransforms() {
-        if (this.transforms.isEmpty()) {
-            for (PluginDesc<Transformation<?>> plugin : herder.plugins().transformations()) {
-                if ("org.apache.kafka.connect.runtime.PredicatedTransformation".equals(plugin.className())) {
-                    this.transforms.add(new TransformsInfo(HasHeaderKey.class.getName(), (new HasHeaderKey<>().config())));
-                    this.transforms.add(new TransformsInfo(RecordIsTombstone.class.getName(), (new RecordIsTombstone<>().config())));
-                    this.transforms.add(new TransformsInfo(TopicNameMatches.class.getName(), (new TopicNameMatches<>().config())));
+    private synchronized void initTransformsAndPredicates() {
+        if (null == this.transforms || this.transforms.isEmpty()) {
+            final List<TransformDefinition> transformPlugins = new ArrayList<>();
+            final List<PredicateDefinition> predicatePlugins = new ArrayList<>();
+            Herder herder = getHerder();
+            for (PluginDesc<Transformation<?>> transformPlugin : herder.plugins().transformations()) {
+                if ("org.apache.kafka.connect.runtime.PredicatedTransformation".equals(transformPlugin.className())) {
+                    for (PluginDesc<Predicate<?>> predicate : herder.plugins().predicates()) {
+                        predicatePlugins.add(PredicateDefinition.fromPluginDesc(predicate));
+                    }
                 }
                 else {
-                    this.transforms.add(new TransformsInfo(plugin));
+                    TransformDefinition transformDefinition = TransformDefinition.fromPluginDesc(transformPlugin);
+                    if (null != transformDefinition) {
+                        transformPlugins.add(transformDefinition);
+                    }
                 }
             }
+            this.predicates = Collections.unmodifiableList(predicatePlugins);
+            this.transforms = Collections.unmodifiableList(transformPlugins);
         }
-
-        return Collections.unmodifiableList(this.transforms);
     }
 
-    @GET
-    @Path("/topic-creation")
-    public boolean getTopicCreationEnabled() {
-        return this.isTopicCreationEnabled;
-    }
-
-    private synchronized Boolean isTopicCreationEnabled() {
+    private synchronized Boolean isTopicCreationEnabled(Map<String, ?> config) {
         Version kafkaConnectVersion = parseVersion(AppInfoParser.getVersion());
         String topicCreationProperty = (String) config.get("topic.creation.enable");
```

Comment:
Good point

Reply:
So `Map<String, ?>` is just another way to write `Map<String, ? extends Object>`; therefore you can simply recast the map to `Map<String, Object>` temporarily to use the `getOrDefault` method.

Reply:
Isn't it more performant to cast the single retrieved value of type `?` to `String` and compare it like this instead of casting the whole map to <String, Object> only to retrieve a single entry that even might not be set in most cases?